src/intelligence_layer/core/model.py

```diff
@@ -157,6 +157,22 @@ def __init__(
         )
         self._explain = _Explain(self._client, name)
 
+    @property
+    @lru_cache(maxsize=1)
+    def context_size(self) -> int:
```

Comment:
Should we make this a model property instead of an AA-Model property? I believe that would make the models more interchangeable. I'm not super sure about all the downsides of this tho